setup.py

```diff
@@ -29,11 +29,35 @@
     author_email='gilv@il.ibm.com',
     packages=find_packages(),
     install_requires=[
-        'Click', 'pandas', 'PyYAML', 'python-dateutil',
-        'pika==0.13.1', 'glob2', 'tqdm', 'lxml',
-        'tblib', 'docker', 'requests', 'seaborn', 'paramiko',
-        'matplotlib', 'kubernetes', 'ibm-cos-sdk', 'gcsfs',
-        'aliyun-fc2', 'oss2',
+        'Click',
+        'pandas',
+        'PyYAML',
+        'python-dateutil',
+        'pika==0.13.1',
+        'enum34',
+        'glob2',
+        'tqdm',
+        'lxml',
+        'tblib',
+        'docker',
+        'requests',
+        'seaborn',
+        'paramiko',
+        'matplotlib',
+        'kubernetes',
+        'ibm-cos-sdk',
+        'gcsfs',
+        'aliyun-fc2',
+        'oss2',
+        'redis',
+        'boto3',
+        'httplib2',
+        'google-cloud-storage',
+        'google-cloud-pubsub',
+        'google-api-python-client',
+        'google-auth',
+        'azure-storage-blob==2.1.0',
+        'azure-storage-queue==2.1.0'
     ],
```

Comment:
@aitorarjona @JosepSampe we already discussed it in the past. We should avoid that setup install all those dependencies, azure, amazon, google, etc. It should be per backend may be or configuration options in setup, so it's not installed by default at least, only if requested, like setup.py azure, etc.

Reply:
How about moving these dependencies to `extras_require` in `setup.py`?
Then `pip install lithops` would only install required dependencies and, for example, to install Google Cloud backend it would be `pip install lithops[gcp]`.